workers/unity/Packages/com.improbable.gdk.deploymentlauncher/DeploymentLauncher.cs

```diff
@@ -0,0 +1,334 @@
+                EditorGUILayout.EndToggleGroup();
+                simWorkerLaunchJson = EditorGUILayout.TextField("Config", simWorkerLaunchJson);
+                EditorGUILayout.EndToggleGroup();
+
+                EditorGUI.BeginDisabledGroup(runningLaunchTask != null);
+                if (GUILayout.Button("Launch deployment"))
+                {
+                    runningLaunchTask = TriggerLaunchDeploymentAsync();
+                }
+                EditorGUI.EndDisabledGroup();
+
+                // Deployment manager.
+                EditorGUILayout.Space();
+                EditorGUILayout.LabelField("Deployment List", EditorStyles.boldLabel);
+                EditorGUILayout.BeginHorizontal();
+                EditorGUILayout.PrefixLabel("Deployments:");
+                if (runningStopTask != null)
+                {
+                    EditorGUILayout.LabelField($"stopping '{deploymentList[selectedDeployment].Name}'");
+                }
+                else if (runningListTask != null)
+                {
+                    EditorGUILayout.LabelField("refreshing");
+                }
+                else if (deploymentList != null)
+                {
+                    string[] deploymentNames = new string[deploymentList.Count];
+                    for (var i = 0; i < deploymentList.Count; ++i)
+                    {
+                        deploymentNames[i] = deploymentList[i].Name;
+                    }
+                    selectedDeployment = EditorGUILayout.Popup(selectedDeployment, deploymentNames);
+                }
+                EditorGUILayout.EndHorizontal();
+                EditorGUI.BeginDisabledGroup(runningListTask != null || runningStopTask != null || deploymentList.Count == 0);
+                if (GUILayout.Button("Stop Deployment"))
+                {
+                    runningStopTask = TriggerStopDeploymentAsync(deploymentList[selectedDeployment].Id);
+                }
+                EditorGUI.EndDisabledGroup();
+                EditorGUI.BeginDisabledGroup(runningListTask != null || runningStopTask != null);
+                if (GUILayout.Button("Refresh Deployments"))
+                {
+                    runningListTask = TriggerListDeploymentsAsync();
+                }
+                EditorGUI.EndDisabledGroup();
+            }
+
+            void Update()
+            {
+                if (runningStopTask != null && runningStopTask.IsCompleted)
+                {
+                    try
+                    {
+                        if (runningStopTask.Result)
+                        {
+                            deploymentList.RemoveAt(selectedDeployment);
+                            selectedDeployment = 0;
+                        }
+                        else
+                        {
+                            throw new Exception($"Failed to stop deployment {deploymentList[selectedDeployment].Name} with ID {deploymentList[selectedDeployment].Id}.");
```

Comment:
I'd say `Debug.LogError` is nicer than blowing up the Update loop with an exception (here and below)